src/hiccup/compiler.clj

```diff
@@ -73,11 +82,26 @@
   (or content
       (and (html-mode?) (not (void-tags tag)))))
 
+(defn parse-tag
+  [^String tag]
+  (let [id-index    (let [index (.indexOf tag 35)] (when (pos? index) index))
+        class-index (let [index (.indexOf tag 46)] (when (pos? index) index))]
```

Comment:
Maybe `#=(int \#)` and `#=(int \.)` would be easier to read than 35 and 36.

Reply:
I think you can do it this way:

(.indexOf tag "#")
(.indexOf tag ".")

Reply:
Would that be slower? Since it's looking for a string rather than a single character? Though I guess there could be an optimisation for strings of length 1.

Reply:
Yes. Benchmarks don't show any difference

Reply:
7d589579d5086e6c9f9d14235a6af8cfab4c0948